app/models/framework/definition/transpiler.rb

```diff
@@ -27,8 +28,10 @@ def transpile
         end.tap do |klass|
           klass.const_set('Invoice', entry_data_class(:invoice)) if ast.field_defs(:invoice)
           klass.const_set('Order', entry_data_class(:contract))  if ast.field_defs(:contract)
+          klass.const_set('Other', entry_data_class(:other))     if ast.field_defs(:other)
```

Comment:
A general comment: the `Parser` and `Transpiler` still hard-code the three types of block. It would be harder to make this use `SubmissionEntry::TYPES` and I don't think we should try.